src/main/java/com/academia/library/mapper/OrderDetailsMapper.java

```diff
@@ -0,0 +1,29 @@
+package com.academia.library.mapper;
+
+import com.academia.library.dto.OrderDetailsRequest;
+import com.academia.library.dto.OrderDetailsResponse;
+import com.academia.library.model.OrderDetail;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+import java.util.List;
+
+@Mapper(componentModel = "spring")
+public interface OrderDetailsMapper {
+
+    OrderDetailsResponse toDto(OrderDetail orderDetail);
+
+    @Named("toEntity")
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "deleted", constant = "false")
+    @Mapping(target = "bookId", expression = "java(orderDetailsRequest.getOrderDetailRequestDto().getBookId())")
```

Comment:
Why do you use this approach? 
Please redo this like in other mappers
target = "bookId", source = "entity.id"

Reply:
Ok, thank